torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,479 @@
+        "s2_autumn_part2.zip": "4fc5e1d5c772421dba553722433ac3b9",
+        "s2_spring_part1.zip": "2a89687d8fafa7fc7f5e641bfa97d472",
+        "s2_spring_part2.zip": "5845dcae0ab3cdc174b7c41edd4283a9",
+        "s2_summer_part1.zip": "73ca8291d3f4fb7533636220a816bb71",
+        "s2_summer_part2.zip": "5b5816bbd32987619bf72cde5cacd032",
+        "s2_winter_part1.zip": "ca958f7cd98e37cb59d6f3877573ee6d",
+        "s2_winter_part2.zip": "e7aacb0806d6d619b6abc408e6b09fdc",
+    }
+
+    mask_cmap = {1: (0, 153, 0), 0: (255, 255, 255)}
+
+    wc_cmap = {
+        10: (0, 160, 0),
+        20: (150, 100, 0),
+        30: (255, 180, 0),
+        40: (255, 255, 100),
+        50: (195, 20, 0),
+        60: (255, 245, 215),
+        70: (255, 255, 255),
+        80: (0, 70, 200),
+        90: (0, 220, 130),
+        95: (0, 150, 120),
+        100: (255, 235, 175),
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        modality: list[str] = ["mask", "esa_wc", "viirs", "s2_summer"],
+        split: str = "train",
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new MapInWild dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            modality: the modality to download. Choose from: "mask", "esa_wc",
+                "viirs", "s1", "s2_temporal_subset", "s2_[season]".
+            split: one of "train", "validation", or "test"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if ``split`` argument is invalid
+            ImportError: if pandas is not installed
+        """
+        assert split in ["train", "validation", "test"]
+
+        self.checksum = checksum
+        self.root = root
+        self.transforms = transforms
+        self.modality = modality
+        self.download = download
+
+        try:
+            import pandas as pd  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "pandas is not installed and is required to use this dataset"
+            )
+
+        modality.append("split_ids")
+
+        for modal in modality:
+            for modality_link in self.modality_urls[modal]:
+                os.path.join(self.url, modality_link)
+                if modality_link.endswith("csv"):
+                    self._verify(url=modality_link, md5=None)
+                else:
+                    self._verify(
+                        url=modality_link,
+                        md5=self.md5s[os.path.split(modality_link)[1]],
```

Comment:
If you use -1 instead of 1 it will work even for the csv file which isn't in a subdir